### DISCUSSION TITLE
Git rebase infinite loop

### DISCUSSION THREAD

**Discussion by jpgros** [community/community#33037] [New to GitHub]
Hello,
I've done a git rebase to 'merge' 2 branchs on the same.
It would have been simplier to do a merge directly but it's nott he problem.

I keep stuck into a loop where I do 
git add file.txt
git rebase --continue

And avec the git rebase --continue the file is un-added and says I should add or rm it.
Add or rm does the same thing
Git rebase --continue --abort or --skip gets me to the same point.
I'm fine if I get back to the start but I don't know how.
Does someone has a suggestion to help me ?

thanks for your time

**Comment by airtower-luna** [comment]
That's impossible to answer without knowing a lot more about your repository. A good start would be the command you used to start the rebase, and `git status` output in that "stuck" situation.

**Reply by jpgros** [reply]
Yes I understand but I can't show the project is confidential.
Maybe if you have ideas, the problem seems to bet relatively basic

**Reply by airtower-luna** [reply]
Oh wait, it really is pretty simple. :smile_cat: 

> git add file.txt
> git rebase --continue
> And avec the git rebase --continue the file is un-added and says I should add or rm it.

With Git add you've *staged* the file. You still need to *commit* it before continuing the rebase. If you want to change the current commit instead of inserting one, use `git commit --amend`.

See also [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), or if I'm guessing correctly that you prefer French from the "avec" the translated version [Utilitaires Git - Réécrire l’historique](https://git-scm.com/book/fr/v2/Utilitaires-Git-R%C3%A9%C3%A9crire-l%E2%80%99historique).

**Reply by jpgros** [reply]
I found the problem it came from how I was quitting vi with a q!
And we need to save the file to validate the change (e.g ZZ).

**Reply by queenofcorgis** [reply]
Nice work @jpgros 🤩 and thanks @airtower-luna for jumping in 😄
